live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/storage/StorageClientIntegrationTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.storage;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.S3;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import org.alfresco.hxi_connector.live_ingester.storage.local.LocalStorageClient;
+import org.alfresco.hxi_connector.live_ingester.storage.local.LocalStorageConfig;
+import org.alfresco.hxi_connector.live_ingester.utils.DockerTags;
+
+@SpringBootTest(classes = {LocalStorageConfig.class, SignedStorageClient.class})
+@ActiveProfiles({"test"})
+@Testcontainers
+class StorageClientIntegrationTest
```

Comment:
I wonder if we could separate unit and integration tests by adding a new test source, like here: https://github.com/Alfresco/hxi-connector/tree/feature/add-it-test. It is really easy to configure and can be really handy as you don't mix unrelated stuff. wdyt @krdabrowski @tpage-alfresco @mpichura?

Reply:
I was also thinking about that and so far, I don't see any contraindications 🤔 I can adjust it, but let see what rest will say

Reply:
Separating `src/test/java` from `src/integration-test/java` is fine. Doing a quick scan of our existing projects then it seems more standard to rename the other way - into `src/unit-test/java` and `src/test/java`.

Reply:
can we separate it then?

Reply:
It's definitely worth separating out different types of tests.
Although I usually used to do that with packages (unit tests in same packages as tested classes, integration tests in separate package), I am okay with separate test sources as this at times be even more effective.

Reply:
Ok, I will work on that - I will go with `src/test/java` and `src/integration-test/java` as it's more popular convention in Java world. Also I think, that unit test are these primary kind of tests and they shouldn't be offended with some prefixes/suffixes.

Reply:
After small investigation it turned out, that it's not a trivial stuff to do - Maven is designed this way, that all test should be placed in one folder - of course there exist a way to workaround it, but not without issues. In general the community recommendations are: leave all kinds of tests in one (dedicated by Maven) folder, or create a separate Maven module for that: e.g. https://stackoverflow.com/questions/37998040/where-should-the-integration-tests-be-stored-when-using-the-maven-failsafe-plugi or https://stackoverflow.com/questions/46831999/maven-add-integration-tests and plenty more...
I think this issue deserves a separate Jira task, so won't be solved here. FYI @Pawel-608 @tpage-alfresco @mpichura

Reply:
Definitely fine to raise a separate ticket for this to be done. I found several projects with `src/unit-test/java` so I think we can probably copy something from one of those to set the configuration.

Reply:
Task created: https://alfresco.atlassian.net/browse/ACS-6514